src/mainloop.rkt

```diff
@@ -328,24 +328,26 @@
   (define original-points (setup-context! vars (or specification prog) precondition repr))
   (run-improve! iters prog specification preprocess original-points repr))
 
-
 (define (make-preprocess-pcontext prog pcontext iters
                                   #:specification [specification #f]
                                   #:preprocess [preprocess '()])
   (timeline-event! 'preprocess)
 
   ;; If the specification is given, it is used for sampling points
-  (define sortable (connected-components specification (*context*)))
-
-  (define new-preprocess
-    (for/list ([sortable-variables (in-list sortable)]
+  (define preprocess-sort
+    (for/list ([sortable-variables (connected-components specification (*context*))]
                #:when (> (length sortable-variables) 1))
       (cons 'sort sortable-variables)))
-  (timeline-push! 'symmetry (map ~a new-preprocess))
-  (*herbie-preprocess* (append preprocess new-preprocess))
+  (define preprocess-abs
+    (for/list ([absable-variable (get-fabs specification (*context*))])
+      (list 'abs absable-variable)))
 
-  (preprocess-pcontext pcontext (*herbie-preprocess*) (*context*)))
+  (timeline-push! 'symmetry (map ~a preprocess-sort))
```

Comment:
In short, I think you can do `(timeline-push! 'symmetry (map ~a preprocess-abs))`.